pages/docs/tutorials/native/interop-with-c.md

```diff
@@ -0,0 +1,140 @@
+---
+type: tutorial
+layout: tutorial
+title:  "Interop with C Libraries"
+description: "A look at how to interop with C libraries"
+authors: Hadi Hariri 
+date: 2018-01-09
+showAuthorInfo: false
+---
+
+
+When writing native applications, oftentimes we need to access certain functionality that is not included in the Kotlin standard library, 
+such as making HTTP calls, reading and writing from disk, etc.
+
+Kotlin/Native provides us with the ability to consume standard C libraries, opening up an entire ecosystem of functionality that exists 
+for pretty much anything we could need. 
+
+In this tutorial we'll see how to create a native Kotlin application that uses the `libcurl` C library to call an HTTP endpoint. We'll cover
+how to 
+
+* [Create Kotlin Bindings](#generating-bindings)
+* [Consume a generated Kotlin API](#consuming-the-kotlin-api)
+* [Link the library into the application](#compiling-and-linking-the-library)
+
+
+## Generating Bindings
+
+An ideal scenario for interop is to call C functions as if we were calling Kotlin functions, that is, following the same signature and conventions. And that is precisely what the 
+`cinterop` tool provides us with. It takes a C library and generates the corresponding Kotlin bindings for it, which then allow us
+to use the library as if it were Kotlin code. 
+
+In order to generate these bindings, we need to create a library definition file (`def`) that contains some information about the headers we need to generate. In our case we want to use the famous `libcurl` library
+to make some HTTP calls, so we'll create a file named `libcurl.def` with the following contents
+
+```
+headers = curl/curl.h
+headerFilter = curl/*
+linkerOpts.osx = -L/opt/local/lib -L/usr/local/opt/curl/lib -lcurl
+linkerOpts.linux = -L/usr/lib/x86_64-linux-gnu -lcurl```
+```
+
+A few things are going on in that file so let's see them one by one. The first entry is `headers` which is the list of header files that we want to generate 
+Kotlin stubs for. We can add multiple files to this entry, separating each one with `\` on a new line. In our case we only want `curl.h`. The files we reference
+need to be relative to the folder where the definition file is, or be available on the system path (in our case it would be `/usr/bin/curl`).
+
+The second line is the `headerFilter`. This is used to denote what exactly we want included. In C, when one file references another file with the `#include` directive, 
+all the headers are also included. Sometimes this may not be needed, and we can use this parameter, [using glob patterns](https://en.wikipedia.org/wiki/Glob_(programming)), to fine tune things. 
```

Comment:
Maybe add something like:
"Note, that headerFilter is an optional argument, required if library you're using is being installed as a system library and you do not want to fetch external dependencies (such as system stdint.h header) into your interoperability library. It may be important for both library size optimizations and fixing potential conflicts between system and Kotlin/Native provided compilation environment."